site/docs/expressions/lambda_expressions.md

```diff
@@ -0,0 +1,137 @@
+# Lambda Expressions
+
+Lambda expressions represent inline, anonymous functions within query plans, enabling operations that require nested computation over data. They are primarily used to express array operations (such as `transform` or `filter`) and support higher-order functions that apply computations element-wise over collections. A lambda consists of explicit parameter types and a body expression that references those parameters.
+
+## Overview
+
+Lambda expressions are a type of expression in Substrait (like `IfThen`, `Subquery`, or `Nested` expressions) that can be passed as arguments to higher-order functions or [invoked directly](#lambda-invocation).
+
+!!! note "Documentation Syntax"
+    This documentation uses the syntax `(param: type, ...) -> expression` as an illustrative notation to explain lambda concepts in a readable form. There is no formal syntax specified in the Substrait spec for compactly representing lambdas.
+
+## Lambda Expression Structure
+
+A lambda expression consists of:
+
+| Component          | Description                                                                 | Protobuf Field        | Required |
+|--------------------|-----------------------------------------------------------------------------|-----------------------|----------|
+| Parameters         | Struct type defining the lambda's parameters. Each field in the struct represents a parameter that can be accessed via FieldReference. The struct's nullability must be `NULLABILITY_REQUIRED`. | `parameters`     | Yes      |
+| Body Expression    | The expression to evaluate (can reference parameters via LambdaParameterReference). The type of this expression is the return type of the lambda. | `body`                | Yes      |
```

Comment:
Minor nit.  In the first row you say "a parameter that can be accessed via FieldReference".  In the second row you say "reference parameters via LambdaParameterReference".  I get what you are saying (a LambdaParameterReference is a type of field reference) but it is slightly confusing.

Maybe change the first instance to "a parameter that can be accessed via LambdaParameterReference (a type of FieldReference)" or change the second instance to "can reference parameters with a FieldReference of type LambdaParameterReference)"